relay/http.go

```diff
@@ -38,8 +44,36 @@ func NewHTTP(cfg HTTPConfig) (Relay, error) {
 		if b.Name == "" {
 			b.Name = b.Location
 		}
-
-		h.backends = append(h.backends, &httpBackend{b.Name, b.Location})
+		timeout := DefaultHTTPTimeout
+		if b.Timeout != "" {
+			t, err := time.ParseDuration(b.Timeout)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing HTTP timeout %v", err)
+			}
+			timeout = t
+		}
+		// If configured, create a retryBuffer per backend.
+		// This way we serialize retries against each backend.
+		var rb *retryBuffer
+		if cfg.BufferSize > 0 {
+			max := DefaultMaxRetryTime
+			if cfg.MaxRetryTime != "" {
+				m, err := time.ParseDuration(cfg.MaxRetryTime)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing max retry time %v", err)
+				}
+				max = m
+			}
+			rb = newRetryBuffer(cfg.BufferSize/len(cfg.Outputs), max)
```

Comment:
I think just noting that the buffer size config option will be per-backed should suffice. That way it's easy to reason about how many input batches can be buffered without having to count the number of outputs and divide the buffer size by the count.